src/token-stream.js

```diff
@@ -16,6 +16,21 @@
+
+function isIdentifierPartES6(char) {
+  let charCode = char.codePointAt(0);
+  if (charCode < 0xFFFF && code.isIdentifierPartES6(charCode)) {
+    return true;
+  }
+  if (char.match(IDENTIFIER_CONTINUE) != null) {
```

Comment:
```js
IDENTIFIER_CONTINUE.test(char)
```

Reply:
Actually, what does this even do beyond `code.isIdentifierPartES6`? I think that's sufficient.

Reply:
@michaelficarra `code.isIdentifierPartES6` doesn't support surrogate pairs.